File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int ptr=0, l=nums.length, i=0;
        while (ptr<l && i<l)
        {
            if( nums[i]!=0)
            {
                nums[ptr]= nums[i];
                ptr++;
            }
            i++;
        }
        while(ptr<l)
        {
            nums[ptr++]=0;
        }
    }
}

File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n= len(nums)
        last_non_zero=0
        for current in range(n):
            if nums[current]!=0:
                if current !=last_non_zero:
                    nums[last_non_zero]= nums[current]
                    nums[current]= 0
                last_non_zero+=1